src/ctelnet.cpp

```diff
@@ -743,42 +743,53 @@ bool cTelnet::socketOutRaw(std::string& data)
     return true;
 }
 
-void cTelnet::setDisplayDimensions()
+void cTelnet::checkNAWS()
 {
-    int x = (mpHost->mScreenWidth < mpHost->mWrapAt) ? mpHost->mScreenWidth : mpHost->mWrapAt;
-    int y = mpHost->mScreenHeight;
-    if (myOptionState[static_cast<size_t>(OPT_NAWS)]) {
-        std::string s;
-        s = TN_IAC;
-        s += TN_SB;
-        s += OPT_NAWS;
-        char x1, x2, y1, y2;
-        x1 = x / 256;
-        x2 = x % 256;
-        y1 = y / 256;
-        y2 = y % 256;
-        //IAC must be doubled
-        s += x1;
-        if (x1 == TN_IAC) {
-            s += TN_IAC;
-        }
-        s += x2;
-        if (x2 == TN_IAC) {
-            s += TN_IAC;
-        }
-        s += y1;
-        if (y1 == TN_IAC) {
-            s += TN_IAC;
-        }
-        s += y2;
-        if (y2 == TN_IAC) {
-            s += TN_IAC;
-        }
+    Host* pHost = mpHost;
+    if (!pHost) {
+        return;
+    }
+    int naws_x = (pHost->mScreenWidth < pHost->mWrapAt) ? pHost->mScreenWidth : pHost->mWrapAt;
+    int naws_y = pHost->mScreenHeight;
+    if ((naws_y > 0) && (myOptionState[static_cast<size_t>(OPT_NAWS)]) && ((mNaws_x != naws_x) || (mNaws_y != naws_y))) {
+        qDebug().nospace().noquote() << "cTelnet::checkNAWS()() INFO - want to send NAWS data (for: \"" << mProfileName << "\") of X = " << naws_x << ", Y = " << naws_y << ".";
```

Comment:
@vadi2 How about this one - it needs a typo cleaning up anyhow?
```suggestion
        qDebug().nospace().noquote() << "cTelnet::checkNAWS() INFO - want to send NAWS data (for: \"" << mProfileName << "\") of X = " << naws_x << ", Y = " << naws_y << ".";
```